Source/WebCore/platform/spiel/PlatformSpeechSynthesizerSpiel.cpp

```diff
@@ -0,0 +1,294 @@
+        self->m_platformSynthesizer.client().didStartSpeaking(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "utterance-finished", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielUtterance*) {
+        self->m_platformSynthesizer.client().didFinishSpeaking(*self->m_utterance);
+        self->clearUtterance();
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "utterance-canceled", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielUtterance*) {
+        self->m_platformSynthesizer.client().didFinishSpeaking(*self->m_utterance);
+        self->clearUtterance();
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "utterance-error", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielUtterance*) {
+        self->m_platformSynthesizer.client().speakingErrorOccurred(*self->m_utterance);
+        self->clearUtterance();
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "notify::paused", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielSpeaker* speaker) {
+        gboolean isPaused;
+        g_object_get(speaker, "paused", &isPaused, nullptr);
+        if (isPaused)
+            self->m_platformSynthesizer.client().didPauseSpeaking(*self->m_utterance);
+        else
+            self->m_platformSynthesizer.client().didResumeSpeaking(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "notify::voices", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielSpeaker*) {
+        self->m_platformSynthesizer.client().voicesDidChange();
+    }), this);
+
+    m_speakerCreatedCallback();
+}
+
+SpielSpeechWrapper::~SpielSpeechWrapper()
+{
+    g_signal_handlers_disconnect_by_data(m_speaker.get(), this);
+}
+
+String SpielSpeechWrapper::generateVoiceURI(const GRefPtr<SpielVoice>& voice, const String& language)
+{
+    auto provider = spiel_voice_get_provider(voice.get());
+    return makeString(URI_PREFIX, span(spiel_provider_get_well_known_name(provider)), '#', span(spiel_voice_get_identifier(voice.get())), '#', language);
+}
+
+Vector<RefPtr<PlatformSpeechSynthesisVoice>> SpielSpeechWrapper::initializeVoiceList()
+{
+    Vector<RefPtr<PlatformSpeechSynthesisVoice>> platformVoices;
+    auto voices = spiel_speaker_get_voices(m_speaker.get());
+    unsigned position = 0;
+    m_voices.clear();
+    while (auto item = g_list_model_get_item(voices, position)) {
+        auto voice = SPIEL_VOICE(item);
+        auto name = makeString(span(spiel_voice_get_name(voice)));
+        const char* const* languages = spiel_voice_get_languages(voice);
+        for (unsigned i = 0; i < G_N_ELEMENTS(languages); i++) {
+            auto language = makeString(span(languages[i]));
+            bool isDefault = !i;
+            auto uri = generateVoiceURI(voice, language);
+            platformVoices.append(PlatformSpeechSynthesisVoice::create(uri, name, language, false, isDefault));
+            m_voices.add(uri, GRefPtr(voice));
+        }
+        position++;
```

Comment:
Move this `position++` into the while clause above. It is not used anywhere else.